pkg/controllers/provisioningv2/cluster/controller.go

```diff
@@ -190,15 +190,34 @@ func (h *handler) generateProvisioningClusterFromLegacyCluster(cluster *v3.Clust
 	if !h.isLegacyCluster(cluster) || cluster.Spec.FleetWorkspaceName == "" {
 		return nil, status, nil
 	}
-	return []runtime.Object{
-		&v1.Cluster{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:        cluster.Name,
-				Namespace:   cluster.Spec.FleetWorkspaceName,
-				Labels:      yaml.CleanAnnotationsForExport(cluster.Labels),
-				Annotations: yaml.CleanAnnotationsForExport(cluster.Annotations),
-			},
+	provCluster := &v1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        cluster.Name,
+			Namespace:   cluster.Spec.FleetWorkspaceName,
+			Labels:      yaml.CleanAnnotationsForExport(cluster.Labels),
+			Annotations: yaml.CleanAnnotationsForExport(cluster.Annotations),
 		},
+	}
+
+	if cluster.Spec.ClusterAgentDeploymentCustomization != nil {
+		clusterCopy := cluster.Spec.ClusterAgentDeploymentCustomization.DeepCopy()
```

Comment:
Nit, but `clusterCopy` to me reads as full cluster object copy. Can we just go with `copy` or if want to be more verbose as `clusterAgentCustomizationCopy` (very verbose lol).